debian/rules

```diff
@@ -128,6 +128,22 @@ DEPENDS += bcc-tools, \
 %:
 	dh $@
 
+override_dh_install:
+	#
+	# All the files that we want to include in this package have to
+	# be present in the "stage" directory. Here we move them into 
+	# debian/tmp since it is a special directory and dh_install will
+	# strip "debian/tmp" from the destination path when --auto-dest
+	# is set. 
+	#
+	# Note that the debian/tmp/ directory is cleaned up when
+	# dpkg-buildpkg is ran so we cannot use debian/tmp/ instead of
```

Comment:
Can you explain this restriction?

I think it might be cleaner to just use "debian/tmp" instead of "stage", moving the complexity that's currently in `Makefile` to here. E.g.
```
-   mv stage/* debian/tmp/
+
+   cp -r files/common/* debian/tmp/
+   if [ -d files/$(TARGET_PLATFORM)/ ]; then \
+       cp -r files/$(TARGET_PLATFORM)/* debian/tmp/; \
+   fi
+
+   sed "s/@@TARGET_PLATFORM@@/$(TARGET_PLATFORM)/" \
+       debian/tmp/var/lib/delphix-appliance/platform.in \
+       >debian/tmp/var/lib/delphix-appliance/platform
+   rm debian/tmp/var/lib/delphix-appliance/platform.in
+
+   ./scripts/download-signature-key.sh upgrade 5.3 \
+       >debian/tmp/var/lib/delphix-appliance/key-public.pem.upgrade.5.3
+
    dh_install --autodest "debian/tmp/*"
```

In general, I don't like polluting the filesystem with temporary files and directories, like we'd now be doing with the "stage" directory, when it can be avoided. Further, when running `make packages`, we'd clobber the "stage" directory in between each different platform packages build (right?), so I don't see any benefit to using a new "stage" directory over "debian/tmp" directly.

Reply:
Yeah I think moving the Makefile logic into `debian/rules` is cleaner. I'll give it a try.

The restriction was that we can't move files into debian/tmp directly from the Makefile, since debian/tmp would be clobbered when dpkg-buildpacakge is invoked.